lib/ghost_adapter/config.rb

```diff
@@ -92,21 +92,30 @@ def compact
       to_h.compact
     end
 
-    def as_args
-      with_env.map { |key, value| arg(key, value) }.compact
+    def as_args(context: {})
```

Comment:
starting to think the whole "args" logic should move from `Config` and into `Command`

Reply:
I think that moving at least the ENV vars into the command section makes sense. At the moment understanding the priority order of the different ways to configure options is the most complex part of this library, it would be great to be able to simplify that, out of scope for this PR however.

Reply:
any chance you wanna make an issue for this and describe the desired outcome?  I think you're on to something with ENV config not being merged into config _within_ the config class, but instead the two being merged just-in-time when the command is formed.